packages/mobile/src/navigator/Headers.tsx

```diff
@@ -47,31 +48,57 @@ export const headerWithCancelButton = {
   headerLeft: <CancelButton />,
 }
 
-export const exchangeHeader = (makerToken: CURRENCY_ENUM, makerTokenBalance: string) => {
+interface Props {
+  title: string
+  token: CURRENCY_ENUM
+}
+
+export function HeaderTitleWithBalance({ title, token }: Props) {
+  const dollarBalance = useSelector((state) => state.stableToken.balance)
+  const goldBalance = useSelector((state) => state.goldToken.balance)
+
+  const balance = token === CURRENCY_ENUM.GOLD ? goldBalance : dollarBalance
+
+  return (
+    <View style={styles.header}>
+      {title && <Text style={fontStyles.headerTitle}>{title}</Text>}
+      <Text style={fontStyles.subSmall}>
+        {balance != null ? (
+          <Trans i18nKey="balanceAvailable" ns={Namespaces.global}>
+            <CurrencyDisplay
+              amount={{
+                value: balance,
+                currencyCode: CURRENCIES[token].code,
+              }}
+            />{' '}
+            available
+          </Trans>
+        ) : (
+          i18n.t('global:loading')
```

Comment:
Not sure we can be certain the balance is loading just because it's null here. Maybe just leave a TODO or comment here mentioning that?

Reply:
Agreed!